src/opera/pge/dswx_pge.py

```diff
@@ -222,6 +223,127 @@ def _geotiff_filename(self, inter_filename):
         # TODO: include band once SAS produces per-band outputs
         return f"{core_filename}.tif"
 
+    def _collect_dswx_product_metadata(self):
+        """
+        Gathers the available metadata from a sample output DSWx product for
+        use in filling out the ISO metadata template for the DSWx-HLS PGE.
+
+        Returns
+        -------
+        output_product_metadata : dict
+            Dictionary containing DSWx-HLS output product metadata, formatted
+            for use with the ISO metadata Jinja2 template.
+
+        """
+        # Find a single representative output DSWx-HLS product, they should all
+        # have identical sets of metadata
+        output_products = self.runconfig.get_output_product_filenames()
+        sample_output_product = None
```

Comment:
This is just a nit, but mapbe representative_output_product rather than sample_output_product.